docs/frameworks/vitepress.md

```diff
@@ -4,34 +4,258 @@ title: VitePress | Frameworks
 
 # VitePress
 
-Since [VitePress](https://vitepress.vuejs.org/) is built using [Vuejs 3](https://v3.vuejs.org/) and 
-on top of [Vite](https://vitejs.dev/), the integration with this plugin is graceful.
+::: warning
+We recommend you use the latest version of VitePress. The latest versions will also require you to update your application to use Vite ^3.1.0.
+:::
 
-## Plugin Configuration
+::: info
+For `Type declarations`, `Prompt for update` and `Periodic SW Updates` go to [Vue 3](/frameworks/vue#vue-3) entry.
+:::
 
-Just follow the [Getting Started](/guide/) section and use one of the `registerType` option to configure
-the behavior:
+## VitePress PWA Module
 
-- [Prompt for update](/guide/prompt-for-update): prompt for new content refreshing
-- [Automatic reload](/guide/auto-update) when new content available
+`vite-plugin-pwa` provides the new `withPwa` module augmentation that will allow you to use `vite-plugin-pwa` in your VitePress applications.
+
+You will need to install `@vite-pwa/vitepress` using:
+```shell
+pnpm add -D @vite-pwa/vitepress
+```
+
+To update your project to use the new `vite-plugin-pwa` for VitePress, you only need to wrap your VitePress config with `withPwa` (you don't need oldest `pwa` and `pwa-configuration` modules):
+```ts
+// .vitepress/config.ts
+import { defineConfig } from 'vitepress'
+import { withPwa } from '@vite-pwa/vitepress'
+
+export default withPwa(defineConfig({
+  /* your VitePress options */
+  /* Vite PWA Options */
+  pwa: {}
+}))
+```
 
 ## Import Virtual Modules
 
-You need to configure a `custom theme` and use the `Layout` component to register the `ReloadPrompt.vue` component
-when using `prompt for new content available`, see [Vue 3](/frameworks/vue#vue-3).
+Since VitePress uses SSR/SSG, we need to call the `vite-plugin-pwa` virtual module using a dynamic `import`.
```

Comment:
This text is repeated multiple times. Perhaps we can simplify the structure like this:

## Import Virtual Module

Since VitePress uses SSR/SSG, we need to call the `vite-plugin-pwa` virtual module using a dynamic `import`. This can be done in the [theme](https://vitepress.vuejs.org/guide/theme-introduction). You can either configure the plugin to auto update or prompt for update. Refer below for examples.

### Auto Update

::: details .vitepress/theme/index.ts
::: details .vitepress/theme/components/ReloadPrompt.vue

### Prompt for update

::: details .vitepress/theme/index.ts
::: details .vitepress/theme/components/RegisterSW.vue

Reply:
@brc-dd done